libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -475,25 +477,27 @@ bool AC_Fence::check_fence_alt_min()
 /// auto enable fence floor
 bool AC_Fence::auto_enable_fence_floor()
 {
-    if (_floor_enabled) {
-        return true; // already enabled
-    }
-
     // altitude fence check
     if (!(_enabled_fences & AC_FENCE_TYPE_ALT_MIN)) {
         // not enabled
         return false;
     }
 
-    AP::ahrs().get_relative_position_D_home(_curr_alt);
-    _curr_alt = -_curr_alt; // translate Down to Up
+    float alt;
+    AP::ahrs().get_relative_position_D_home(alt);
+    _curr_alt = -alt; // translate Down to Up
 
     // check if we are over the altitude fence
-    if (_curr_alt >= _alt_min + _margin) {
-        enable_floor();
-        gcs().send_text(MAV_SEVERITY_INFO, "Fence floor enabled (auto enabled)");
+    if (!_floor_enabled && _curr_alt >= _alt_min + _margin) {
+        auto_enable_fence_after_takeoff();
 
         return true;
+    } 
+    // check if we are under the altitude fence
+    else if (_floor_enabled && _curr_alt <= _alt_min - _margin) {
+        auto_disable_fence_for_landing();
```

Comment:
this does a  lot more than disable floor (ie disables ALL fences) depending on what autoenablvalue there is e(Plane only again since you dropped the autoenable param) ....you need a auto_disable_floor_for_landing function called from modes rather than this here I think

then in Copter you need to decide in the landing modes whether you want to disable ALL fences (which is now called with auto_disable_fence_for_landing, but always currently does nothing...and still does nothing with this PR...but could be made to do something) or actually make those calls to disable the floor (ie  new auto_disable_floor_for landing)

Reply:
Fixed thanks